src/OmniSharp.Host/Services/AssemblyLoader.cs

```diff
@@ -38,17 +38,26 @@ public IReadOnlyList<Assembly> LoadAllFrom(string folderPath)
         {
             if (string.IsNullOrWhiteSpace(folderPath)) return Array.Empty<Assembly>();
 
-            var assemblies = new List<Assembly>();
-            foreach (var filePath in Directory.EnumerateFiles(folderPath, "*.dll"))
+            try
             {
-                var assembly = LoadFromPath(filePath);
-                if (assembly != null)
+                var assemblies = new List<Assembly>();
+
+                foreach (var filePath in Directory.EnumerateFiles(folderPath, "*.dll"))
                 {
-                    assemblies.Add(assembly);
+                    var assembly = LoadFromPath(filePath);
+                    if (assembly != null)
+                    {
+                        assemblies.Add(assembly);
+                    }
                 }
-            }
 
-            return assemblies;
+                return assemblies;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"An error occurred when attempting to access '{folderPath}'.");
```

Comment:
I had it in the initial commit (https://github.com/OmniSharp/omnisharp-roslyn/pull/866/commits/76906f90f7a2041504a54ffcb8e4f2cd774a59aa) but then I decided to go for try/catch instead as that would cover a lot more potential edge cases i.e. `PathTooLongException`, `SecurityException`, `UnauthorizedAccessException`, `IOException` rather than just `DirectoryNotFoundException`